client/dashboard/domains/domain-forwarding/form.tsx

```diff
@@ -0,0 +1,229 @@
+import {
+	Card,
+	CardBody,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	Button,
+	Panel,
+	PanelBody,
+	PanelRow,
+	RadioControl,
+} from '@wordpress/components';
+import { DataForm } from '@wordpress/dataviews';
+import { useState, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { isTargetUrlValid, isSubdomainValid } from './utils';
+import type { DomainForwarding } from '../../data/domain-forwarding';
+import type { Field } from '@wordpress/dataviews';
+
+export interface FormData {
+	sourceType: 'root' | 'subdomain';
+	subdomain: string;
+	targetUrl: string;
+	redirectType: 'temporary' | 'permanent';
+	pathForwarding: 'no' | 'yes';
+}
+
+interface DomainForwardingFormProps {
+	domainName: string;
+	initialData?: DomainForwarding | null;
+	onSubmit: ( data: FormData ) => void;
+	isSubmitting: boolean;
+	submitButtonText: string;
+}
+
+export default function DomainForwardingForm( {
+	domainName,
+	initialData,
+	onSubmit,
+	isSubmitting,
+	submitButtonText,
+}: DomainForwardingFormProps ) {
+	const [ formData, setFormData ] = useState< FormData >( () => {
+		if ( ! initialData ) {
+			return {
+				sourceType: 'subdomain',
+				subdomain: '',
+				targetUrl: '',
+				redirectType: 'temporary',
+				pathForwarding: 'no',
+			};
+		}
+		const protocol = initialData.is_secure ? 'https://' : 'http://';
+		return {
+			sourceType: initialData.subdomain ? 'subdomain' : 'root',
+			subdomain: initialData.subdomain || '',
+			targetUrl: `${ protocol }${ initialData.target_host }${ initialData.target_path || '' }`,
+			redirectType: initialData.is_permanent ? 'permanent' : 'temporary',
+			pathForwarding: initialData.forward_paths ? 'yes' : 'no',
+		};
+	} );
+
+	const redirectTypeField = {
+		id: 'redirectType',
+		label: __( 'HTTP Redirect Type' ),
+		type: 'text' as const,
+		Edit: 'radio',
+		elements: [
+			{
+				label: __( 'Temporary redirect (307)' ),
+				value: 'temporary',
```

Comment:
Is it possible to make the type of field as `boolean` and change the value to true/false? If so, we can simply name it as `is_permanent` without  any transform  🤔

Reply:
Yes - no idea why I decided to do it this way 🤣

Reply:
So turns out that the `RadioControl` component works with strings but still using the same names as from the API and boolean logic is better